src/main/java/shef/servlets/GroupchatServlet.java

```diff
@@ -17,23 +17,53 @@
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
 
-@WebServlet("/groupchat")
+@WebServlet("/load-groupchat")
 public class GroupchatServlet extends HttpServlet {
 
+  DatastoreService datastore;
+
+  @Override
+  public void init() {
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String keyString = request.getParameter("key");
+    Entity groupchatEntity = null;
+    try {
+      groupchatEntity = datastore.get(KeyFactory.stringToKey(keyString));
+    } catch (EntityNotFoundException e) {
+      e.printStackTrace();
+      return;
+    }
 
+    ArrayList<String> messages = (ArrayList<String>) groupchatEntity.getProperty("messages");
```

Comment:
y arraylist vs. linkedlist?

Reply:
Nothing is going to be added to this list as it's just for returning, so I chose an ArrayList here.

Reply:
lovely